### PR TITLE
feature/vbmc_conf_template

### PR DESCRIPTION
**Pull request by turtle-fly**
[  ] Enable CBMC
[->] Update jinja template of vbmc.conf with more customization
[  ] Enable socat
[  ] Enable task schedule to start/stop socat, bmc, qemu
[  ] Scrub infrasim.yml to avoid duplicated definition

**Enable more customization for bmc feature**

    bmc:
	    interface
		lancontrol
		chassiscontrol
		startcmd
		startnow
		poweroff_wait
		kill_wait
		emu_file
		config_file
		ipmi_over_lan_port
		historyfru

		username
		password

Please check etc/vbmc.conf.example for detail explaination.

**Added test**
1. Unit test for serveral BMC attributes
2. Functional test for IOL port customization

### CODE REVIEW

**Comment by emcinfrasim** [comment]
```
******* Functional Test ******* 
TestName: test_set_cpu_family 
ErrorMessage: exceptions.AssertionError 
ErrorINfo:   File "/usr/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/home/infrasim/workspace/Infrasim_Compute_pull_request_trigger/infrasim-compute/test/functional/test_configuration_change.py", line 58, in test_set_cpu_family
    assert "-cpu IvyBridge" in str_result
'\n-------------------- >> begin captured logging << --------------------\nroot: DEBUG: qemu-system-x86_64 -vnc :1 -name quanta_d51 -device sga --enable-kvm -smbios file=/usr/local/etc/infrasim/quanta_d51/quanta_d51_smbios.bin -boot ncd -machine q35,usb=off,vmport=off -chardev socket,id=mon,host=127.0.0.1,port=2345,server,nowait  -mon chardev=mon,id=monitor -serial mon:udp:127.0.0.1:9003,nowait -uuid 5872d0a6-15c8-412a-b490-ae5821ca4f95 -chardev socket,id=ipmi0,host=127.0.0.1,port=9002,reconnect=10 -device ipmi-bmc-extern,chardev=ipmi0,id=bmc0 -device isa-ipmi-kcs,bmc=bmc0 -net user -net nic -device ahci,id=ahci0 -drive file=/home/infrasim/.infrasim/sda.img,format=qcow2,if=none,id=drive0,cache=writeback -device ide-hd,bus=ahci0.0,drive=drive0 -m 1024 -cpu IvyBridge,+vmx -smp 2,sockets=2,cores=1,threads=1  2>/var/log/qemu.log &\nroot: INFO: qemu start\n--------------------- >> end captured logging << ---------------------' 
```

**Comment by emcinfrasim** [comment]
```
******* Functional Test ******* 
TestName: test_set_cpu_family 
ErrorMessage: exceptions.AssertionError 
ErrorINfo:   File "/usr/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/home/infrasim/workspace/Infrasim_Compute_pull_request_trigger/infrasim-compute/test/functional/test_configuration_change.py", line 58, in test_set_cpu_family
    assert "-cpu IvyBridge" in str_result
'\n-------------------- >> begin captured logging << --------------------\nroot: DEBUG: qemu-system-x86_64 -vnc :1 -name quanta_d51 -device sga --enable-kvm -smbios file=/usr/local/etc/infrasim/quanta_d51/quanta_d51_smbios.bin -boot ncd -machine q35,usb=off,vmport=off -chardev socket,id=mon,host=127.0.0.1,port=2345,server,nowait  -mon chardev=mon,id=monitor -serial mon:udp:127.0.0.1:9003,nowait -uuid 093112a9-959e-4f46-a24f-b46669bdd227 -chardev socket,id=ipmi0,host=127.0.0.1,port=9002,reconnect=10 -device ipmi-bmc-extern,chardev=ipmi0,id=bmc0 -device isa-ipmi-kcs,bmc=bmc0 -net user -net nic -device ahci,id=ahci0 -drive file=/home/infrasim/.infrasim/sda.img,format=qcow2,if=none,id=drive0,cache=writeback -device ide-hd,bus=ahci0.0,drive=drive0 -m 1024 -cpu IvyBridge,+vmx -smp 2,sockets=2,cores=1,threads=1  2>/var/log/qemu.log &\nroot: INFO: qemu start\n--------------------- >> end captured logging << ---------------------' 
```

**Comment by turtle-fly** [comment]
retest this please

**Comment by turtle-fly** [comment]
@sharkconi @xiar @fub2 @XiaowenJiang @PaynePei 

Please review this code change about: BMC customizations and test.

**Comment by turtle-fly** [comment]
Make "historyfru" a customizable variables, with default value 10, according to @xiar 's advice.

File: infrasim/model.py [hunk 1]
```diff
@@ -808,45 +816,142 @@ def __init__(self, bmc_info={}):
...
+            raise ArgsNotCorrect("SOL device {} doesn\'t exist".
+                                 format(self.__sol_device))
+
+        # check attribute
+        if self.__poweroff_wait < 0:
+            raise ArgsNotCorrect("poweroff_wait is expected to be >= 0, "
+                                 "it's set to {} now".
+                                 format(self.__poweroff_wait))
+
+        if type(self.__poweroff_wait) is not int:
```
Comment (xiar) [inline infrasim/model.py]: isinstance(self.__poweroff_wait, int) is better?

**Comment by sharkconi** [comment]
+1

File: infrasim/model.py [hunk 2]
```diff
@@ -808,45 +816,142 @@ def __init__(self, bmc_info={}):
...
+
+        if not os.path.isfile(self.__startcmd_script):
+            raise ArgsNotCorrect("startcmd script {} doesn\'t exist".
+                                 format(self.__chassiscontrol_script))
+
+        # check ports are in use
+        # check lan interface exists
+
+        # check if sol device exists
+        if not os.path.islink(self.__sol_device):
```
Comment (fub2) [inline infrasim/model.py]: :+1:

**Comment by fub2** [comment]
ship it!

**Pull request merged**